src/main/java/com/jcabi/beanstalk/maven/plugin/AbstractBeanstalkMojo.java

```diff
@@ -259,6 +332,85 @@ protected boolean isGreen(final Environment env) {
     }
 
     /**
+     * Validate given text in YAML format.
+     *
+     * @param text YAML text
+     */
+    protected void validYaml(final String text) {
+        new Yaml().load(text);
+    }
+
+    /**
+     * Validate the war file.
+     *
+     * @param zip Zip file
+     * @throws org.apache.maven.plugin.MojoFailureException Thrown, if
+     *  validation fails.
+     */
+    protected void validate(final ZipFile zip) throws MojoFailureException {
+        final ZipEntry ebextdir = zip.getEntry(".ebextensions");
+        if (ebextdir == null) {
+            throw new MojoFailureException(
+                ".ebextensions directory does not exist in the WAR file"
+            );
+        }
+        final Enumeration<? extends ZipEntry> entries = zip.entries();
+        while (entries.hasMoreElements()) {
+            final ZipEntry entry = entries.nextElement();
+            if (entry.getName().startsWith(".ebextensions/")
+                && !entry.isDirectory()) {
+                final String text = this.readFile(zip, entry);
+                try {
+                    this.validYaml(text);
+                } catch (final YAMLException yamlException) {
+                    final String msg =
+                        "File '%s' in .ebextensions is not YAML valid. %s";
+                    throw new MojoFailureException(
+                        String.format(
+                            msg,
+                            entry.getName(),
+                            yamlException.getMessage()
+                        ),
+                        yamlException
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Reads text from a ZIP file.
+     * @param warfile ZIP file, which contains entry.
+     * @param entry ZIP entry (compressed file) to read from.
+     * @return Text content of entry.
+     * @throws MojoFailureException thrown when encounter error.
+     */
+    private String readFile(final ZipFile warfile, final ZipEntry entry)
+        throws MojoFailureException {
+        String text = null;
```

Comment:
@dmzaytsev  If we make it final, we can't change it later on, which means that we can't try catch the IOException and throws a MojoExecutionException and also "finally" close the io to avoid leak. Unless i propagate IOException. I think readFile should handle it's own generated exception and not propagate it If things are marked final in this case, how would we manage the try catch?